goalgorilla/open_social
# Issue #2930486 Posts are orphaned when a user is deleted

The proper hooks are not implemented for the Post entity type to ensure posts are correctly reassigned an author or deleted if a user is removed. 

This pull request implements those methods by looking at how it was done for Nodes.

How to test:
- Ensure /admin/content/post does not contain any errors.
- Create a post in the stream as User1
- Delete User1 from the platform
- Observe that the /admin/content/post page now throws an error
- Apply this PR and run updates
- Observe that /admin/content/post no longer throws an error. The post for User1 has now been deleted.
- Create a post in the stream as User2
- Delete User2
- Depending on the site settings the post created as User2 has been deleted or assigned to the anonymous user. Confirm this by looking at /admin/content/post
